lib/src/components/NxCheckbox/stateful/__tests__/NxStatefulCheckbox.test.tsx

```diff
@@ -4,57 +4,112 @@
+    expect(input).toHaveAttribute('id', 'boo');
+    expect(input).toHaveAttribute('name', 'boo');
   });
 
-  it('sets the initial value of isChecked to the value of defaultChecked prop', function() {
-    const shallowRenderWithDefaultCheckedFalse = getShallow({defaultChecked: false});
-    expect(shallowRenderWithDefaultCheckedFalse).toHaveProp('isChecked', false);
+  it('sets ref on the top-level element', function() {
+    const ref = React.createRef<HTMLLabelElement>(),
+        el = renderEl({ ref });
 
-    const shallowRenderWithDefaultCheckedTrue = getShallow({defaultChecked: true});
-    expect(shallowRenderWithDefaultCheckedTrue).toHaveProp('isChecked', true);
+    expect(ref.current).toBe(el);
   });
 
-  it('adds the nx-checkbox--disabled class if disabled is set', function() {
-    expect(getShallow()).not.toHaveProp('disabled');
-    expect(getShallow({ disabled: true })).toHaveProp('disabled');
+  it('disables the input iff disabled is set', function() {
+    expect(quickRender().getByRole('checkbox')).not.toBeDisabled();
+    expect(quickRender({ disabled: undefined }).getByRole('checkbox')).not.toBeDisabled();
+    expect(quickRender({ disabled: false }).getByRole('checkbox')).not.toBeDisabled();
+    expect(quickRender({ disabled: true }).getByRole('checkbox')).toBeDisabled();
   });
 
-  it('updates isChecked prop on NxCheckbox when the checkbox is toggled', function() {
-    const component = getShallow(simpleProps);
-    expect(component).toHaveProp('isChecked', false);
-    component.simulate('change');
-    expect(component).toHaveProp('isChecked', true);
+  it('renders label text iff there are children', function() {
+    expect(renderEl({ children: 'foo' })).toHaveTextContent('foo');
+    expect(renderEl()).toHaveTextContent('');
   });
 
-  it('calls its onChange prop when the input fires a change event', function() {
-    const onChange = jest.fn(),
-        component = getShallow({ onChange });
+  it('adds the tm-checked class when checked, and the tm-unchecked class when unchecked', async function() {
+    const user = userEvent.setup(),
+        el = renderEl({ children: 'foo' })!,
+        input = within(el).getByRole('checkbox');
+
+    expect(el).toHaveClass('tm-unchecked');
+    expect(el).not.toHaveClass('tm-checked');
+
+    await user.click(input);
+
+    expect(el).toHaveClass('tm-checked');
+    expect(el).not.toHaveClass('tm-unchecked');
+  });
+
+  it('sets the input to checked per the value of defaultChecked', function() {
+    expect(quickRender({ defaultChecked: false }).getByRole('checkbox')).not.toBeChecked();
+    expect(quickRender({ defaultChecked: true }).getByRole('checkbox')).toBeChecked();
+  });
+
+  it('updates checked state when the checkbox is toggled', async function() {
+    const user = userEvent.setup(),
+        input = quickRender().getByRole('checkbox');
+
+    expect(input).not.toBeChecked();
+
+    await user.click(input);
+
+    expect(input).toBeChecked();
```

Comment:
Might as well make sure it toggles back too

Reply:
Updated here db2b7763ab07a078d3b9cedf9335868acdd77ccf. Thanks.